include/api/CDataProcessor.h

```diff
@@ -71,10 +71,14 @@ class API_EXPORT CDataProcessor : private core::CNonCopyable {
                               core_t::TTime& completeToTime) = 0;
 
     //! Persist current state
-    virtual bool persistState(core::CDataAdder& persister) = 0;
+    virtual bool persistState(core::CDataAdder& persister,
+                              const std::string& descriptionPrefix = "") = 0;
```

Comment:
Do we need a default here?  It seems like the default makes the logging far less useful so it would be better to enforce one.